### PR TITLE
[DDW-596] webpack 5 upgrade

### PR DESCRIPTION
**Pull request by danielmain**
This PR updates Webpack 4 to 5 and introduces some DX improvements.

## Todos

- [x] Upgrade Webpack and related dependencies (Dominik)
- [x] Get Webpack 5 to build our project (Dominik)
- [x] Improve `yarn dev` mode with react fast-refresh (Dominik)
- [x] Fix small bug with maintaining window size on page reloads (Dominik)
    How to reproduce: move or resize the app window and reload the page with cmd + R -> the window jumps back into the previous place
- [x] Fix broken test environment (Dominik)
- [x] Update sass-loader and move from `node-sass` (deprecated) to `dart-sass` (Przemysław Włodek)
- [x] fix broken storybook (Dominik)
- [x] fix failing tests (Dominik)
- [x] Make CI builds pass (@michalrus )

## Screenshots

- React fast-refresh for "instant" CSS changes:

https://user-images.githubusercontent.com/172414/145424583-b2f333b3-0a12-43b7-8870-c47aa2ecbcaf.mp4

- React fast-refresh for "instant" component changes:

https://user-images.githubusercontent.com/172414/145425033-5b258a87-4736-46fc-988d-1a49f47c5099.mp4

- Any changes in the renderer outside of components will result in a page reload like previously.
- Any changes in the main process files will reload the whole app

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1653926375311239)
- [ ] Since this changes our build process it potentially affects all areas of our codebase and thorough QA testing is required
- [ ] Especially: Test if maintaining window size feature still works the same as before (+ bug with reloading page after change is fixed)

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [ ] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +53,8 @@ If you get SSL error when running `nix-shell` (SSL peer certificate or SSH remot
 
 1. Run `yarn nix:selfnode` from `daedalus`.
 2. Run `yarn dev` from the subsequent `nix-shell` (use `KEEP_LOCAL_CLUSTER_RUNNING` environment variable to keep the local cluster running after Daedalus exits: `KEEP_LOCAL_CLUSTER_RUNNING=true yarn dev`)
-3. Once Daedalus has started and has gotten past the loading screen run the following commands from a new terminal window if you wish to import funded wallets:
+3. Run `yarn start` from a second `nix-shell` instance (running step 1 from a terminal instance)
```
Comment (renanvalentin) [inline README.md:56]: Whats the main usage of `yarn start`? My current workflow is `yarn nix:testnet && yarn dev`, is it any different?

File: source/renderer/app/components/loading/syncing-connecting/LogosDisplay.scss [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@use 'sass:math';
```
Comment (renanvalentin) [inline source/renderer/app/components/loading/syncing-connecting/LogosDisplay.scss:1]: We could also use CSS calc() to reduce dependency from SASS.

File: source/renderer/app/components/navigation/Navigation.scss [hunk 3]
```diff
@@ -1,3 +1,5 @@
+@use 'sass:math';
```
Comment (renanvalentin) [inline source/renderer/app/components/navigation/Navigation.scss:1]: We could also use CSS calc() to reduce dependency from SASS.

File: .babelrc [hunk 4]
```diff
@@ -5,8 +5,10 @@
     "@babel/preset-flow"
   ],
   "plugins": [
+    ["@babel/plugin-transform-runtime"],
```
Comment (renanvalentin) [inline .babelrc:8]: We could also use `babel-preset` as it loads the appropriate transformers based on the target environment.

**Comment by renanvalentin** [comment]
Huge jump in performance compared to existing solution. Well done 🚀

File: README.md (hunk 1 above)
Comment (DominikGuzei) [reply README.md:56]: `yarn start` can be used after `yarn build` to test a production build

**Comment by DominikGuzei** [comment]
@danielmain can you check the latest comments from @renanvalentin and handle these as well?

File: source/renderer/app/assets/images/spinner-dark-big.inline.svg [hunk 5]
```diff
@@ -1,3 +1,3 @@
-<svg xmlns="http://www.w3.org/2000/svg" width="88" height="88" viewBox="0 0 88 88">
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 88 88">
```
Comment (renanvalentin) [inline source/renderer/app/assets/images/spinner-dark-big.inline.svg:1]: @mchappell do you think `asset/source` loader could've added side-effects to other svgs file? 

Screen Shot 2022-01-05 at 17 33 52

File: storybook/.babelrc [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (renanvalentin) [inline storybook/.babelrc:1]: Looks like the issue on `source/renderer/app/components/wallet/transactions/Transaction.tsx` is related to [babel-plugin-react-intl](formatjs/babel-plugin-react-intl#166). Maybe upgrading to latest version should fix this. 

Also, we could consider removing a few babel plugins in case there are not needed anymore as `preset-typescript` and `preset-env` should cover most of the js spec?

File: source/renderer/app/assets/images/spinner-dark-big.inline.svg (hunk 5 above)
Comment (mchappell) [reply source/renderer/app/assets/images/spinner-dark-big.inline.svg:1]: Good spot! Fixed this in latest push

**Comment by michalrus** [comment]
Hmmm, what is this WebAssembly file? `dist/renderer/7ae194443b56ad7c5250.module.wasm`

**Comment by szymonmaslowski** [comment]
> Hmmm, what is this WebAssembly file? `dist/renderer/7ae194443b56ad7c5250.module.wasm`

Good question... I can add that it is already on develop.

File: shell.nix [hunk 7]
```diff
@@ -90,7 +89,7 @@ let
     NODE_EXE = "cardano-wallet";
     CLI_EXE = "cardano-cli";
     NODE_IMPLEMENTATION = nodeImplementation;
-    BUILDTYPE = "Debug";
+    BUILDTYPE = "Release";
```
Comment (michalrus) [inline shell.nix:92]: Not sure about this change. /cc @danielmain

File: shell.nix [hunk 8]
```diff
@@ -113,11 +112,10 @@ let
       ln -svf $(type -P cardano-wallet)
       ln -svf $(type -P cardano-cli)
       mkdir -p ${BUILDTYPE}/
-      ln -svf $PWD/node_modules/usb/build/${BUILDTYPE}/usb_bindings.node ${BUILDTYPE}/
-      ln -svf $PWD/node_modules/node-hid/build/${BUILDTYPE}/HID.node ${BUILDTYPE}/
-      ln -svf $PWD/node_modules/node-hid/build/${BUILDTYPE}/HID_hidraw.node ${BUILDTYPE}/
+      cp $PWD/node_modules/usb/build/${BUILDTYPE}/usb_bindings.node ${BUILDTYPE}/
```
Comment (michalrus) [inline shell.nix:115]: Why copy instead of link?

File: package.json [hunk 9]
```diff
@@ -5,15 +5,21 @@
...
-    "check:all": "yarn prettier:check && yarn lint && yarn compile && yarn stylelint && yarn lockfile:check && yarn manage:translations && yarn storybook:build",
-    "start": "gulp start",
-    "start:dev": "NODE_ENV=development gulp start",
-    "dev": "IS_WATCH_MODE=true gulp dev",
+    "dev": "concurrently --names renderer,main 'yarn dev:renderer' 'wait-on http://localhost:8080 && yarn dev:main'",
+    "dev:windows": "cross-env DAEDALUS_INSTALL_DIRECTORY=\"C:\\Program Files\\Daedalus Testnet\" LAUNCHER_CONFIG=\"C:\\Program Files\\Daedalus Testnet\\launcher-config.yaml\" yarn dev",
+    "dev:main": "cross-env NODE_ENV=development yarn build:main --watch",
+    "dev:renderer": "concurrently 'yarn typedef:sass --watch' 'cross-env NODE_ENV=development yarn webpack serve -c source/renderer/webpack.config.js --progress'",
+    "build": "yarn build:cleanup && yarn build:main && yarn build:renderer",
+    "build:main": "yarn webpack -c source/main/webpack.config.js --progress --mode production",
```
Comment (michalrus) [inline package.json:13]: Why isn’t `--mode production` picked from environment’s `NODE_ENV=production`?

**Comment by michalrus** [comment]
Okay, seems done

File: scripts/minimize-runtime-dependencies--linux.sh [hunk 10]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+#
+# This script *can* help in minimizing runtime dependencies of
+# Daedalus… but only the ones which prevent Electron from starting.
+#
+# That’s a good starting point before we get this automatically from
+# webpack – <https://input-output.atlassian.net/browse/DDW-1103>.
+#
```
Comment (michalrus) [inline scripts/minimize-runtime-dependencies--linux.sh:9]: I used this script to quickly find missing installer dependencies, perhaps it should be removed from here, and added to `webpack-bundle-analyzer` PR by @szymonmaslowski:
* https://input-output.atlassian.net/browse/DDW-1103
* commit: https://github.com/input-output-hk/daedalus/commit/6b8e610b33ac94ea7d987c34c70a1661bd4cd508
* branch: https://github.com/input-output-hk/daedalus/tree/webpack-bundle-analyzer

File: shell.nix [hunk 11]
```diff
@@ -57,9 +57,6 @@ let
       git python27 curl jq
       nodePackages.node-gyp nodePackages.node-pre-gyp
       gnumake
-      (if system == "aarch64-darwin"
```
Comment (michalrus) [inline shell.nix:60]: Can someone verify this particular change on M1?

File: source/renderer/app/components/wallet/receive/WalletReceiveRandom.tsx [hunk 12]
```diff
@@ -277,27 +277,23 @@ class WalletReceiveRandom extends Component<Props> {
             </div>
 
             {walletAddresses.length ? (
-              <>
```
Comment (michalrus) [inline source/renderer/app/components/wallet/receive/WalletReceiveRandom.tsx:280]: I wonder if those changes are needed, adding to the diff size

File: source/renderer/app/containers/wallet/PaperWalletCreateCertificatePage.tsx [hunk 13]
```diff
@@ -120,16 +120,16 @@ class PaperWalletCreateCertificatePage extends Component<Props, State> {
...
-    // eslint-disable-next-line react/no-access-state-in-setstate
-    const prevStep = this.state.currentStep ? this.state.currentStep - 1 : 0;
-    const prevDialog = this.CREATE_CERTIFICATE_DIALOGS[prevStep];
-    this.setState({
-      currentStep: prevStep,
-    });
-    this.switchDialog(prevDialog);
-    this.props.actions.wallets.updateCertificateStep.trigger(true);
-  };
+  // onBack = () => {
```
Comment (michalrus) [inline source/renderer/app/containers/wallet/PaperWalletCreateCertificatePage.tsx:123]: Hmmmm, why not remove instead of commenting out?

File: source/renderer/index.ejs [hunk 14]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+  <head>
+    <meta charset="utf-8">
+    <!--
+        unsafe-eval is require to run WASM code
```
Comment (michalrus) [inline source/renderer/index.ejs:5]: Why do we have WebAssembly code at all?

I asked this question above, what is this file?

```
dist/renderer/7ae194443b56ad7c5250.module.wasm
```

File: source/main/windows/main.ts [hunk 15]
```diff
@@ -38,12 +38,15 @@ type WindowOptionsType = {
   };
   icon?: string;
 };
-export const createMainWindow = (locale: string, windowBounds?: Rectangle) => {
+export const createMainWindow = (
```
Comment (michalrus) [inline source/main/windows/main.ts:41]: Why isn’t this instance changed to the new (old?) `function createMainWindow(locale…` syntax?

File: source/renderer/app/components/settings/categories/SecuritySettings.tsx [hunk 16]
```diff
@@ -13,13 +13,13 @@ type Props = {
   onOpenDiscreetModeToggle: () => void;
 };
 
-const SecuritySettings = ({
+function SecuritySettings({
```
Comment (michalrus) [inline source/renderer/app/components/settings/categories/SecuritySettings.tsx:16]: Not sure why only some functions change syntax style from lambda to `function`. I suggested we do style changes in a separate PR, but it was rejected. ¯\\\_(ツ)\_/¯

**Review by michalrus — COMMENTED** [review]
This PR seems crazy large, but there are three main offenders, constituting 79.6% of all changes:

```
% git diff --stat develop..@ -- source/renderer/app/i18n/locales/defaultMessages.json translations/messages.json yarn.lock
 source/renderer/app/i18n/locales/defaultMessages.json | 21476 ++++++++++++------------------------------------------
 translations/messages.json                            |  7587 +++++++++++++++++++
 yarn.lock                                             | 12461 +++++++++++++++----------------
 3 files changed, 18318 insertions(+), 23206 deletions(-)
```

Without those three:

```
 270 files changed, 6480 insertions(+), 4167 deletions(-)
```

And further, with `--ignore-all-space`, just 15.6% of the original:

```
 269 files changed, 5222 insertions(+), 2908 deletions(-)
```

That is still rather large, but more manageable… I have not reviewed those three files.

File: source/main/windows/main.ts (hunk 15 above)
Comment (DominikGuzei) [reply source/main/windows/main.ts:41]: Good question, need to check if that is somehow excluded from the checks

File: source/renderer/app/components/settings/categories/SecuritySettings.tsx (hunk 16 above)
Comment (DominikGuzei) [reply source/renderer/app/components/settings/categories/SecuritySettings.tsx:16]: Long story short: we started updating/fixing linting issues that have been introduced to package upgrades (which might not have been necessary) … now the question is: do we want to revert those (legitimate but unhelpful) changes just to reduce the size of this PR, creating duplicate work at other places?

File: source/renderer/app/components/wallet/summary/WalletSummaryAsset.tsx [hunk 17]
```diff
@@ -0,0 +1,209 @@
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/renderer/app/components/wallet/summary/WalletSummaryAsset.tsx:1]: good question, I have no idea 👍 maybe there was another feature PR merged into this webpack 5 branch because it required webpack 5? That's kinda the only reason i can think of right now

**Review by gabriela-ponce — APPROVED** [review]

**Review by dmitrii-gaico — APPROVED** [review]

File: source/renderer/app/components/wallet/summary/WalletSummaryAsset.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/renderer/app/components/wallet/summary/WalletSummaryAsset.tsx:1]: Those files have been removed from this PR as they have been moved to other locations on develop and thus would just create duplication.

File: source/main/windows/main.ts (hunk 15 above)
Comment (DominikGuzei) [reply source/main/windows/main.ts:41]: @michalrus i think that's because they other changes are required by some react-specific linting rule that suggests using named functions because it is then easier to debug react components in the browser (because they have names that show up in the React developer tools etc.)

File: source/main/windows/main.ts [hunk 18]
```diff
@@ -144,13 +151,13 @@ export const createMainWindow = (locale: string, windowBounds?: Rectangle) => {
    * window constructor above was buggy (height was not correctly applied)
    */
   window.on('ready-to-show', () => {
-    if (windowBounds) {
-      window.setBounds(windowBounds);
+    const savedWindowBounds = getSavedWindowBounds();
+    if (savedWindowBounds) {
+      window.setBounds(savedWindowBounds);
```
Comment (DominikGuzei) [inline source/main/windows/main.ts:156]: These changes are part of the bug fix regarding window bounds

File: source/main/windows/main.ts [hunk 19]
```diff
@@ -77,7 +80,11 @@ export const createMainWindow = (locale: string, windowBounds?: Rectangle) => {
     if (event.sender !== window.webContents) return;
     window.close();
   });
-  window.loadURL(`file://${__dirname}/../renderer/index.html`);
+  if (isDev) {
+    window.loadURL(`http://localhost:8080`);
```
Comment (DominikGuzei) [inline source/main/windows/main.ts:84]: this is necessary because in develop we are now using webpack hot server

**Review by renanvalentin — CHANGES_REQUESTED** [review]
Really amazing job!

File: source/main/webpack.config.js [hunk 20]
```diff
@@ -1,106 +1,105 @@
...
-            process.env.MOCK_TOKEN_METADATA_SERVER_PORT || 0,
-          'process.env.MOBX_DEV_TOOLS': process.env.MOBX_DEV_TOOLS || 0,
-          'process.env.BUILD_NUMBER': JSON.stringify(
-            process.env.BUILD_NUMBER || 'dev'
-          ),
-          'process.env.IS_WATCH_MODE': process.env.IS_WATCH_MODE === 'true',
-          'process.env.KEEP_LOCAL_CLUSTER_RUNNING':
-            process.env.KEEP_LOCAL_CLUSTER_RUNNING === 'true',
-        },
+        {},
```
Comment (lucas-barros) [inline source/main/webpack.config.js:102]: It seems this plugin is no longer needed, should we remove it?

File: source/renderer/app/containers/MenuUpdater/MenuUpdater.ts [hunk 21]
```diff
@@ -1,12 +1,11 @@
 import { inject, observer } from 'mobx-react';
```
Comment (lucas-barros) [inline source/renderer/app/containers/MenuUpdater/MenuUpdater.ts:1]: `inject` is now unused

File: source/renderer/app/i18n/locales/en-US.json [hunk 22]
```diff
@@ -36,6 +38,12 @@
   "appUpdate.overlay.postponeInstall.link.label": "Postpone the update",
   "appUpdate.overlay.subtitle": "You are currently running Daedalus version <b>{currentAppVersion}</b>.<br />Daedalus version <b>{availableAppVersion}</b> is now available to download.",
   "appUpdate.overlay.title": "Software update available!",
+  "asset.transactionConfirmation.assetLabel": "!!!Token #{assetNumber}",
```
Comment (lucas-barros) [inline source/renderer/app/i18n/locales/en-US.json:41]: There translations were removed automatically after I removed this unused [code](https://github.com/input-output-hk/daedalus/pull/2956/files#diff-d89b13bb59bd2d2a3412173a5c8d62a25607cb7672096154e28846d4c28d8d5fL98). Maybe they came back with those wrongly added wallet token file. We might need to run "manage:translations" again?

**Review by miorsufianiohk — APPROVED** [review]
LGTM on all 4 builds: intel Mac, M1 Mac, Linux and Windows. Tested on 21914. Great work everyone 👍 (ps: Great job on fixing the M1 build @michalrus )

File: scripts/minimize-runtime-dependencies--linux.sh (hunk 10 above)
Comment (DominikGuzei) [reply scripts/minimize-runtime-dependencies--linux.sh:9]: @michalrus what's the status on this?

File: source/common/ipc/lib/IpcChannel.ts [hunk 23]
```diff
@@ -84,7 +84,8 @@ export class IpcChannel<Incoming, Outgoing>
 
     // Enforce the singleton pattern based on the channel name
     const existingChannel = IpcChannel._instances[channelName];
-    if (existingChannel) return existingChannel;
+    if (existingChannel)
+      throw new Error(`Channel ${channelName} already exists`);
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/common/ipc/lib/IpcChannel.ts:88]: This is fixing an eslint error:
grafik

File: source/main/preload.ts [hunk 24]
```diff
@@ -58,10 +47,12 @@ process.once('loaded', () => {
   }
 
   // ESLint will warn about any use of eval(), even this one
-  // eslint-disable-next-line no-eval
-  global.eval = () => {
-    throw new Error('This app does not support window.eval().');
-  };
+  if (environment.isProduction) {
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/main/preload.ts:50]: To enable eval-based source maps from webpack (which are the fastest/best option)

File: source/main/webpack.config.js [hunk 25]
```diff
@@ -1,106 +1,105 @@
...
+      stdio: 'inherit',
+      shell: true,
+    });
+    this.isRunning = true;
+    // Handle next electron shutdown
+    this._process.once('close', () => this.onProcessClose());
+  }
+  restart() {
+    this._shouldRestart = true;
+    this._process.kill();
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/main/webpack.config.js:20]: Yes, it's a standard node.js child process created with [spawn](https://nodejs.org/api/child_process.html#subprocesskillsignal)

File: source/main/windows/main.ts (hunk 19 above)
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/main/windows/main.ts:84]: Because that's the webpack dev server - but i agree we could pass this as env var into the main process 👍 but i would do that in a separate PR tbh

File: source/renderer/app/stores/AppUpdateStore.ts [hunk 26]
```diff
@@ -177,8 +177,9 @@ export default class AppUpdateStore extends Store {
   // @ts-ignore ts-migrate(2749) FIXME: 'News' refers to a value, but is being used as a t... Remove this comment to see the full error message
   _checkNewAppUpdate = async (update: News) => {
     const { version, url } = this.getUpdateInfo(update);
-    // @ts-ignore ts-migrate(1320) FIXME: Type of 'await' operand must either be a valid pro... Remove this comment to see the full error message
-    const appUpdateCompleted = await this.getAppUpdateCompletedRequest.execute();
+    const appUpdateCompleted =
+      // @ts-ignore ts-migrate(1320) FIXME: Type of 'await' operand must either be a valid pro... Remove this comment to see the full error message
+      await this.getAppUpdateCompletedRequest.execute();
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/renderer/app/stores/AppUpdateStore.ts:182]: yeah looks like it 👍 this is auto formatted by prettier

File: source/main/webpack.config.js (hunk 20 above)
Comment (DominikGuzei) [reply source/main/webpack.config.js:102]: yes probably 👍 would you like to setup a Jira ticket for this?

File: source/renderer/app/i18n/locales/en-US.json (hunk 22 above)
Comment (DominikGuzei) [reply source/renderer/app/i18n/locales/en-US.json:41]: Ok thanks, i will update to latest develop and manage translations 👍

File: source/renderer/index.ejs [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+    >
+    <title>Daedalus</title>
+    <script>
+      var global = global || window;
+    </script>
+  </head>
+  <body>
+    <div id="root"></div>
+  </body>
+</html>
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/renderer/index.ejs:26]: you mean `.html` right?

File: source/renderer/app/i18n/locales/en-US.json (hunk 22 above)
Comment (DominikGuzei) [reply source/renderer/app/i18n/locales/en-US.json:41]: @lucas-barros but these translations are still used by components like https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/components/assets/AssetTransactionConfirmation.tsx#L21 … i can see that there are no translations provided for them though … not sure if these components are still used somewhere? If not then we should probably rather delete them and not just the i18n keys (which are auto-generated)

File: scripts/minimize-runtime-dependencies--linux.sh (hunk 10 above)
Comment (michalrus) [reply scripts/minimize-runtime-dependencies--linux.sh:9]: This will be very elegantly superseded by @szymonmaslowski’s https://github.com/input-output-hk/daedalus/pull/2997 :bow: 

Thanks for catching, I’ll drop this file from here! :bow:

File: shell.nix (hunk 11 above)
Comment (michalrus) [reply shell.nix:60]: I.e. start `yarn dev` on an M1

**Comment by michalrus** [comment]
Thank you, everyone! :bow: :sparkles:

**Comment by DominikGuzei** [comment]
A miracle happened 🎉 thanks everyone for your precious contributions!!
